platforms/jvm/plugins-application/src/main/java/org/gradle/jvm/application/tasks/CreateStartScripts.java

```diff
@@ -147,85 +157,66 @@ protected JavaModuleDetector getJavaModuleDetector() {
     /**
      * The environment variable to use to provide additional options to the JVM.
      */
-    @Nullable
     @Optional
     @Input
-    public String getOptsEnvironmentVar() {
-        if (GUtil.isTrue(optsEnvironmentVar)) {
-            return optsEnvironmentVar;
-        }
-
-        if (!GUtil.isTrue(getApplicationName())) {
-            return null;
-        }
-
-        return GUtil.toConstant(getApplicationName()) + "_OPTS";
+    @ReplacesEagerProperty
+    public Property<String> getOptsEnvironmentVar() {
+        return optsEnvironmentVar;
     }
 
     /**
      * The environment variable to use to control exit value (Windows only).
      */
-    @Nullable
     @Optional
     @Input
-    public String getExitEnvironmentVar() {
-        if (GUtil.isTrue(exitEnvironmentVar)) {
-            return exitEnvironmentVar;
-        }
-
-        if (!GUtil.isTrue(getApplicationName())) {
-            return null;
-        }
-
-        return GUtil.toConstant(getApplicationName()) + "_EXIT_CONSOLE";
+    @ReplacesEagerProperty
+    public Property<String> getExitEnvironmentVar() {
+        return exitEnvironmentVar;
     }
 
     /**
      * Returns the full path to the Unix script. The target directory is represented by the output directory, the file name is the application name without a file extension.
+     * TODO: This should be Provider[RegularFile], but we don't support such upgrade with @ReplacesEagerProperty
      */
     @Internal
-    public File getUnixScript() {
-        return new File(getOutputDir(), getApplicationName());
+    @ReplacesEagerProperty(replacedAccessors = @ReplacedAccessor(value = GETTER, name = "getUnixScript"))
+    public RegularFileProperty getUnixScript() {
+        return getObjectFactory().fileProperty().value(
+            getOutputDir().zip(getApplicationName(), Directory::file)
+        );
     }
```

Comment:
❓ Should we add that kind of upgrade?

Reply:
We could add auto upgrade for simple cases. 
This particular case is a bit more complex, since it combines two properties. But wecan handle that now with simpler custom interceptors, I will update the code. 

What makes automatic upgrades a bit harder is one know issue if property has `@OutputFile` annotation: https://github.com/gradle/gradle/issues/29335

Reply:
In any case the code as is is wrong. We should never create new instances of `Property` in a getter.